### PR TITLE
intake: Clean up and fix names

### PR DESCRIPTION
**Pull request by auscompgeek**
The terrible naming was _really_ getting on my nerves.

Looking at IntakeAutomation, I don't think the `on_enable` part is the right place to put that... but I'm not sure what a nice solution for that would look like.

### CODE REVIEW

File: automations/intake.py [hunk 1]
```diff
@@ -1,80 +1,72 @@
...
 
     @state(first=True, must_finish=True)
     def intake_cube(self, state_tm):
-        """Starts the intake motors while waiting for the cube be seen by the
-        infrared sensor"""
-        if self.intake.seeing_cube() and state_tm > 0.5:
-            self.intake.intake_clamp(False)
+        """Start the intake motors and wait for the infrared sensor to see the cube."""
+        if self.intake.is_seeing_cube() and state_tm > 0.5:
+            self.intake.clamp(False)
```
Comment (ArthurAllshire) [inline automations/intake.py]: Do we not want to also pull in the kicker in this state?

File: components/intake.py [hunk 2]
```diff
@@ -31,61 +33,40 @@ def on_disable(self):
...
+        """Get the distance of the infrared sensor in m."""
+        return self.infrared.getDistance() / 100
 
-    def seeing_cube(self):
-        """Returns True when the infrared sensor reads between 0.1m to 0.15m."""
-        if 0.1 <= self.cube_distance <= 0.15:
-            return True
-        else:
-            return False
+    def is_seeing_cube(self) -> bool:
```
Comment (ArthurAllshire) [inline components/intake.py]: I think that this name is still fairly non-descriptive - its not checking whether we are seeing the cube, it is checking that the cube is in the containment mech.

File: components/intake.py [hunk 3]
```diff
@@ -31,61 +33,40 @@ def on_disable(self):
...
+        """Check whether the infrared sensor reads between 0.1m to 0.15m."""
+        return 0.1 <= self.get_cube_distance() <= 0.15
 
-    def contacting_cube(self):
-        """Returns True of the current output of the motor is above 5."""
-        if self.intake_left.getOutputCurrent() >= 5:
-            return True
-        else:
-            return False
+    def is_contacting_cube(self) -> bool:
```
Comment (ArthurAllshire) [inline components/intake.py]: What is contacting the cube? This name is still misleading as the intake is still contacting the cube after we suck it in, but this function will return `False`.

File: automations/intake.py (hunk 1 above)
Comment (auscompgeek) [reply automations/intake.py]: That sounds like a good idea.

File: components/intake.py [hunk 4]
```diff
@@ -31,61 +33,40 @@ def on_disable(self):
...
-        """Turns the extension pneumatics on or off."""
+    def extend(self, value: bool):
+        """Turn the extension pneumatics on or off."""
         self.extension_on = value
 
-    def infrared_distance(self):
-        """Gets the distance of the infrared sensor in m."""
-        self.cube_distance = self.infrared.getDistance() / 100
-        return self.cube_distance
+    def get_cube_distance(self) -> float:
```
Comment (ArthurAllshire) [inline components/intake.py]: We should really note the failure condition for the infrared somewhere (you were saying something about it returning 35 the other day?) @Matthew13479

**Review by ArthurAllshire — CHANGES_REQUESTED** [review]

File: components/intake.py (hunk 2 above)
Comment (auscompgeek) [reply components/intake.py]: `is_cube_contained` sounds like a much better name. I really wasn't sure what this was actually being used for from the name nor the docstring here. Will fix.

File: components/intake.py (hunk 3 above)
Comment (auscompgeek) [reply components/intake.py]: Well, at least the docstring actually describes what this is actually used for now.

What would be a more appropriate and concise name for this method?

File: automations/intake.py [hunk 5]
```diff
@@ -1,80 +1,72 @@
...
+        self.intake.clamp(True)
+        self.intake.push(False)
         self.next_state("stop")
 
     @state(must_finish=True)
     def deposit(self):
         """Deposit cube."""
-        if self.intake.seeing_cube():
+        if self.intake.is_seeing_cube():
             self.next_state("push_out_cube")
```
Comment (auscompgeek) [inline automations/intake.py]: Now that I actually know what that `seeing_cube` method was intended to do, this state seems wrong.

File: components/intake.py (hunk 3 above)
Comment (ArthurAllshire) [reply components/intake.py]: Something like `wheels_contacting_cube`?

Comment (auscompgeek) [reply components/intake.py]: It returns a boolean, so it should be phrased in the form of a question. `are_wheels_contacting_cube` seems unwieldy.

Comment (ArthurAllshire) [reply components/intake.py]: What if we made some of these single line functions into properties?

Comment (auscompgeek) [reply components/intake.py]: Not really much of a fan of using a property here. Not sure how expensive the getOutputCurrent call is.

**Review by ArthurAllshire — APPROVED** [review]

**Pull request merged**